### ISSUE TITLE
Deprecated attributed emitted to MaxPool operator

### ISSUE THREAD

**Issue by kraiskil** [onnx/keras-onnx#597]
According to https://github.com/onnx/onnx/blob/master/docs/Operators.md#MaxPool, the auto_pad attribute to onnx operator MaxPool is deprecated. I.e. paddings should be always be explicit (?).
keras2onnx 1.7.0 adds this attribute to a MaxPool node emitted from `keras.layers.MaxPooling2D()`

**Comment by jiafatom** [comment]
The link you provided does not say auto_pad attribute gets deprecated, it says:

auto_pad : string (default is NOTSET)
auto_pad must be either NOTSET, SAME_UPPER, SAME_LOWER or VALID. Where default value is NOTSET, which means explicit padding is used. SAME_UPPER or SAME_LOWER mean pad the input so that the output spatial size match the input.In case of odd number add the extra padding at the end for SAME_UPPER and at the beginning for SAME_LOWER. VALID mean no padding.

This is not deprecated.

**Comment by kraiskil** [comment]
Yes, it would be nice if it was mentioned in the attribute description as well. Now it feels a bit "hidden".

(sorry, I don't know how to perma-link to a line in a .md file - but see last line in this screen shot of master branch docs)
![image](https://user-images.githubusercontent.com/595342/91021918-20eb4e80-e5fd-11ea-9823-41c824535833.png)

**Comment by jiafatom** [comment]
Thanks for pointing this out. It says auto_pad is a deprecated attribute, but there is also an "If you are using them currently", seems still can use. Currently, it seems that our converted model does not have any issues on this. If it really deprecate, the onnx model will throw error saying there is no such attribute when performing inference. Seems it is still good now. 
Do you have any model that has issues?

**Comment by kraiskil** [comment]
Ok. I don't know enough about onnx internals to comment. I'm writing my own backend/compiler, and bumped into a model that used this deprecated feature. And since my compiler is new, I had thought I might skip implementing the deprecated feature.

I checked, and there are even models with MaxPool auto_pad attribute in the onnx backed test suite. So it seems to be a mild deprecation indeed, and I should just implement it :)
